tests/stdlib/trst.nim

```diff
@@ -460,3 +460,21 @@ suite "RST inline markup":
           rnLeaf  ' '
           rnLeaf  'end'
         """)
+
+    # a more complex URL with some made-up ending '&='.
+    # Github Markdown would include final &= and
+    # so would rst2html.py in contradiction with RST spec.
```

Comment:
do you have a link discussing this?

it's such an edge case that it's good enough for this PR, but would be good to link to spec since there's a conflict between commonly used tools and spec

Reply:
Only https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#standalone-hyperlinks
> Punctuation at the end of a URI is not considered part of the URI, unless the URI is terminated by a closing angle bracket (">").

Punctuation characters are not strictly defined but RST spec mentions `&` and `=` among them in https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#sections . It also mentions Unicode "punctuation" category (though `=` does not fall into it)

And `tkPunct` in `rst.nim` has its own list of characters.

Interestingly, vim syntax highlighting parses that string just as this PR.

I think such a case is very rare because URLs almost always have format like `?=word` or `#word` so I would not bother to dig more.

Reply:
BTW `)` and other punctuation characters are allowed in URLs generally. So the mentioned rule is an invention of those markup language for the sake of convenience.

rst2html.py claims that it parses URLs according to the corresponding RFC for each protocol type. (Of course we don't implement that stuff.) Though its behavior seems weird in some cases:

![image](https://user-images.githubusercontent.com/1299583/116760894-e6a13f00-aa1e-11eb-8e3c-a15fc3595deb.png)

Github Markdown is more reasonable here:

See https://www.google.com/url?sa=t&source=web&cd=&cad=rja&url=https%3A%2F%2Fnim-lang.github.io%2FNim%2Frst.html%23features&usg=AO&  end.

I just don't feel like it's a good time investment to dig into all those minor edge cases.

Reply:
thanks for the links, and yes, it's good enough :)